src/Nodes1DProvisioner.cpp

```diff
@@ -37,6 +40,11 @@ Nodes1DProvisioner::Nodes1DProvisioner(int _NOrder, int _NumElements, double _xm
     EToV = new Array<int, 2>(NumElements, NumFaces);
     EToE = new Array<int, 2>(NumElements, NumFaces);
     EToF = new Array<int, 2>(NumElements, NumFaces);
+    Fmask = new Array<int, 1> (NumFacePoints*NumFaces);
+    Fx = new Array<double, 2>(NumFacePoints*NumFaces, NumElements);
+
+    vmapM = new index_vector_type(NumFacePoints*NumFaces*NumElements);
+    vmapP = new index_vector_type(NumFacePoints*NumFaces*NumElements);
 }
```

Comment:
Maybe add some checks that these pointers are not null.

Reply:
I'm not sure if I know what you mean by this. I'm assigning non-null things to these pointers right here in the constructor, right?

Reply:
You are dynamically allocating memory for an object with new. If the memory allocation fails, then either a bad_alloc exception is thrown or a nullptr is returned. In this case, you are using a throwing new operator, in which case an exception is thrown on failure. So it looks like you don't need to check for a nullptr after all.